File: v1/mistral:7b/spanish/JavaScript/song_info.js
const axios = require('axios');

// Tu client ID y client secret desde Spotify Developer Dashboard
const client_id = 'TU_CLIENT_ID';
const client_secret = 'TU_CLIENT_SECRET';

// Función que obtiene la información de una canción utilizando su nombre.
async function getSongInfo(songName) {
    // Autorizar con Spotify
    const auth = btoa(`${client_id}:${client_secret}`);
    const responseAuth = await axios.get('https://accounts.spotify.com/api/token', {
        headers: {
            Authorization: `Basic ${auth}`,
            'Content-Type': 'application/x-www-form-urlencoded'
        },
        params: {
            grant_type: 'client_credentials'
        }
    });

    // Enviar solicitud a la API de Spotify para obtener información sobre la canción
    const responseSong = await axios.get(`https://api.spotify.com/v1/search?q=${songName}&type=track`, {
        headers: {
            Authorization: `Bearer ${responseAuth.data.access_token}`
        }
    });

    // Devolver información sobre la canción
    if (responseSong.data.tracks.items.length > 0) {
        const song = responseSong.data.tracks.items[0];
        console.log(`Artista: ${song.artists[0].name}`);
        console.log(`Género: ${song.genres}`);
    } else {
        console.log('La canción no pudo ser encontrada');
    }
}

// Llama a la función con el nombre de una canción
getSongInfo('Shape of You - Ed Sheeran');

File: v1/mistral:7b/spanish/Python/song_info.py
# ¡Hola! Si quieres buscar el artista y género de una canción utilizando la API de Spotify en Python, puedes usar las librerías `requests` y `beautifulsoup4`. A continuación te proporciono un código sencillo que hace lo que pides:

# ```python
import requests
from bs4 import BeautifulSoup

def get_song_info(title):
    # Obtenemos la URL de la canción en Spotify
    search_url = f"https://api.spotify.com/v1/search?q={title}&type=track&limit=1"
    response = requests.get(search_url)
    soup = BeautifulSoup(response.content, 'json')

    # Obtenemos los detalles de la canción en caso de encontrarla
    if len(soup.find_all("tracks")) > 0:
        track = soup.find_all("tracks")[0]

        # Extraemos el artista y el género de la canción
        artist = track['artists'][0]['name']
        genre = track['album']['genres'][0] if 'genres' in track['album'] else "No está definido"

        print(f"Artista: {artist}, Género: {genre}")
    else:
        print("La canción no fue encontrada")

if __name__ == "__main__":
    title = input("Ingresa el título de la canción: ")
    get_song_info(title)
# ```

# El script anterior toma el nombre de una canción como entrada y busca la información en Spotify. Recuerda instalar las librerías `requests` y `beautifulsoup4` antes de ejecutarlo. Puedes hacerlo con este comando:

# ```
# pip install requests beautifulsoup4
# ```